servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ClosableConcurrentStack.java

```diff
@@ -28,22 +29,56 @@
     @SuppressWarnings("rawtypes")
     private static final AtomicReferenceFieldUpdater<ClosableConcurrentStack, Object> topUpdater =
             newUpdater(ClosableConcurrentStack.class, Object.class, "top");
+    @SuppressWarnings("rawtypes")
+    private static final AtomicIntegerFieldUpdater<ClosableConcurrentStack> sizeUpdater =
+            AtomicIntegerFieldUpdater.newUpdater(ClosableConcurrentStack.class, "size");
     @Nullable
     private volatile Object top;
+    private volatile int size;
+    private final int maxSize;
+
+    /**
+     * Create a new instance.
+     */
+    ClosableConcurrentStack() {
+        this(128);
```

Comment:
Rational for this number is most external use cases are to support `ListenableAsyncClosable` which tend to have a small set of fixed size `Subscriber`. Our internal repeat/retry use cases typically cancel which remove old `Subscriber` so the size doesn't need to be that big in those cases. Open to making this larger since it is a change in behavior (but likely will catch bugs/leaks, and force people to set a higher max value if required).

Reply:
Don't mind changing the default as long as users can understand what is wrong/changed from the popped up error and what they should do. Currently, the exception message from `ClosableConcurrentStack` is too abstract to understand the root cause.